### PR TITLE
feat: Allow df.drop to take an index object

### PR DESCRIPTION
**Pull request by milkshakeiii**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery-dataframes/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

File: bigframes/dataframe.py [hunk 1]
```diff
@@ -925,10 +927,53 @@ def drop(
             )
         if columns:
             block = block.drop_columns(self._sql_names(columns))
-        if not index and not columns:
+        if index is None and not columns:
             raise ValueError("Must specify 'labels' or 'index'/'columns")
         return DataFrame(block)
 
+    def _drop_by_index(self, index: indexes.Index) -> DataFrame:
```
Comment (TrevorBergeron) [inline bigframes/dataframe.py:934]: Can we do this with a block-level join and filter op? Implementing as a block-operation will make it easier to share logic with series. I also worry that this complex index manipulation has edge cases, in particular when it comes to column multi-indices.

File: bigframes/dataframe.py [hunk 2]
```diff
@@ -906,16 +906,18 @@ def drop(
                 columns = labels
 
         block = self._block
-        if index:
+        if index is not None:
             level_id = self._resolve_levels(level or 0)[0]
 
-            if utils.is_list_like(index):
+            if utils.is_list_like(index) or isinstance(index, pandas.Index):
```
Comment (TrevorBergeron) [inline bigframes/dataframe.py:912]: Doesn't utils.is_list_like already match indices?

Comment (milkshakeiii) [reply bigframes/dataframe.py:912]: good point, only the fix above ("if index is not None:") is needed. This line is not.

File: bigframes/dataframe.py (hunk 1 above)
Comment (milkshakeiii) [reply bigframes/dataframe.py:957]: Okay, I have refactored it to use block-level join and filter op. The code should be much improved now, or at least much shortened.

**Review by tswast — APPROVED** [review]

**Pull request merged**